[WIP] Allow forcing host architeture

Blackhex opened the pull request:
This change allows forcing host architecture for the compilation. When running `vcbuild.bat` inside x64 Python shell on Arm64 marchine, Python always overwrites `PROCESSOR_ARCHITECTURE` environment variable with `AMD64`.

Contributes to https://github.com/dotnet-microsoft/ARM-on-Windows-Ecosystem/issues/202

Commit:
Allow forcing host architeture

Blackhex committed:
Revert "Allow forcing host architeture"

This reverts commit 6d26d9d08e343b7e57fb56f6cd0d3f00892e08f4.

Blackhex committed:
Allow forcing host architeture

eukarpov reviewed (commented):
maybe python should we fixed instead then?

Blackhex commented:
Yeah, It would require further investigation on how the `PROCESSOR_ARCHITECTURE=AMD64` gets to the x64 Conda Python interpreter enviroment as it does not seem to be set in CPython codebase. So far we can consider this PR as a temporary workaround.

pimterry committed:
http: keep HTTP/1.1 conns alive even if the Connection header is removed

Previously persistence was completed disabled if you removed this
header, which is not correct for modern HTTP, where the header is
optional and all connections should persist by default regardless.

PR-URL: https://github.com/nodejs/node/pull/46331
Reviewed-By: Matteo Collina <matteo.collina@gmail.com>
Reviewed-By: Minwoo Jung <nodecorelab@gmail.com>
Reviewed-By: James M Snell <jasnell@gmail.com>
Reviewed-By: Ricky Zhou <0x19951125@gmail.com>
